classes/class.ilMumieTaskGradeSync.php

```diff
@@ -205,4 +218,38 @@ private function getLatestGrade($xapi_grades)
         }
         return $latest_grade;
     }
+
+    public function wasGradeOverriden($user_id)
+    {
+        global $ilDB;
+        $hashed_user = ilMumieTaskIdHashingService::getHashForUser($user_id, $this->task);
+        $query = "SELECT new_grade
+        FROM xmum_grade_override
+        WHERE " .
+        "usr_id = " . $ilDB->quote($hashed_user, "text") .
+        " AND " .
+        "task_id = " . $ilDB->quote($this->task->getId(), "integer");
+        $result = $ilDB->query($query);
+        $grade = $ilDB->fetchAssoc($result);
+        return !is_null($grade["new_grade"]);
+    }
+
+    private function getOverridenGrade($user_id, $xapi_grades)
```

Comment:
This function should be moved to another class (something like `ilMumieTaskGradeOverrideService`). IlMumieTaskGradeSync is only responsible for retrieving data from MUMIE/Lemon servers and filtering them.

Reply:
done